Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -902,28 +902,26 @@ AXTextMarker AXTextMarker::findWordOrSentence(AXDirection direction, bool findWo
 
     // Functions to update resultMarker for word and sentence text units.
     auto updateWordResultMarker = [&] () {
-        int start, end;
-        findWordBoundary(flattenedRuns, offset, &start, &end);
-        if (direction == AXDirection::Previous) {
+        if (direction == AXDirection::Previous && boundary == AXTextUnitBoundary::Start) {
             int previousWordStart = findNextWordFromIndex(flattenedRuns, offset, false);
-            findEndWordBoundary(flattenedRuns, previousWordStart, &end);
-
-            // If start (from the forward-search) is the same as the offset, that means we are on a
-            // start word boundary and shouldn't update the text marker.
-            // When looking backward, the end of a word can be at the offset.
-            if (start != Checked<int>(offset) || end == Checked<int>(offset)) {
-                if (boundary == AXTextUnitBoundary::Start && previousWordStart < objectBorder && previousWordStart != -1)
-                    resultMarker = AXTextMarker(*currentObject, previousWordStart, origin);
-                else if (boundary == AXTextUnitBoundary::End && end <= objectBorder && end != -1)
-                    resultMarker = AXTextMarker(*currentObject, end, origin);
+            if (previousWordStart <= objectBorder)
+                resultMarker = AXTextMarker(*currentObject, previousWordStart, origin);
+        } else if (direction == AXDirection::Next && boundary == AXTextUnitBoundary::End) {
+            int nextWordEnd;
+            findEndWordBoundary(flattenedRuns, offset, &nextWordEnd);
+            if (nextWordEnd >= objectBorder) {
```

Comment:
Is there some plain-english comment we use here to describe what `nextWordEnd` being greater than or equal to `objectBorder` means? e.g. "If the word end position is right on the boundary of an object, we need to ..."